src/libs/elektra/keyset.c

```diff
@@ -2472,22 +2473,21 @@ Key * ksLookup (KeySet * ks, Key * key, option_t options)
  */
 Key * ksLookupByName (KeySet * ks, const char * name, option_t options)
 {
-	Key * found = 0;
-
+	// Note: function MUST NOT be recursive !!
 	if (!ks) return 0;
 	if (!name) return 0;
 
 	if (!ks->size) return 0;
 
-	struct _Key key;
+	if (!ks->lookup)
```

Comment:
Did you measure that this improves performance? `keyNew` is quite expensive and having a key on a stack is (hopefully) very cheap.

Reply:
In terms of performance, the expensive part of `keyNew` (without value or metadata) is actually `elektraKeySetName`. That much I have already found through profiling.

But you might actually be right, the stack repeated allocation might be better than a single heap allocation. I assumed `elektraKeySetName` would reuse the name buffer and only `realloc` when needed. In fact it always `free`s and `malloc`s, so reusing a `Key` won't help much.

Reply:
Yes. It might be more beneficial if we directly optimize how keys+values are handled. But this is a huge topic which needs a lot of profiling.